libc/string/string_tokpbrkstr.c

```diff
@@ -0,0 +1,875 @@
+
+	TEST_ASSERT_NOT_NULL(reversStr);
+
+	for (i = 0; i < ASCII_LENGTH - 1; i++) {
+		reversStr[i] = testStr[ASCII_LENGTH - i - 2];
+	}
+	reversStr[ASCII_LENGTH - 1] = 0;
+
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "cba"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "bac"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "bca"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "aa"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "bb"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "cc"));
+	TEST_ASSERT_EQUAL_STRING("abc", strstr("abc", "ab"));
+	TEST_ASSERT_EQUAL_STRING("bc", strstr("abc", "bc"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "ac"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "aabbcc"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "aaaaaa"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "bbbbbb"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "cccccc"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "ccaabb"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "bbaacc"));
+
+	/* Asserting reversed string */
+	TEST_ASSERT_EQUAL_PTR(NULL, strstr(testStr, reversStr));
+
+	free(testStr);
+}
+
+
+TEST(string_str, big_string)
+{
+	int i;
+	char str[PATH_MAX];
+
+	memset(str, 1, sizeof(str));
+
+	for (i = 1; i < PATH_MAX - 1; i++) {
+		str[i] = 2;
+		TEST_ASSERT_EQUAL_PTR(&str[i], strstr(str, "\2"));
+		TEST_ASSERT_EQUAL_PTR(str, strstr(str, "\1"));
+
+		str[i] = 1;
+	}
+}
+
+
+/*
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////
+*/
+
+
+TEST_SETUP(string_pbrk)
+{
+}
+
+
+TEST_TEAR_DOWN(string_pbrk)
+{
+}
+
+
+TEST(string_pbrk, basic)
+{
+	const char loremIpsum[BUFFSIZE] = LOREM_IPSUM;
+	char *asciiStr;
+
+	asciiStr = testdata_createCharStr(ASCII_LENGTH);
+
+	TEST_ASSERT_EQUAL_PTR(loremIpsum, strpbrk(loremIpsum, "Lorem"));
+
+	/*
+	 * strpbrk is sensitive to all elements in the array where
+	 * is looking up which means 'm' is the first element of array 'loremIpsum' that strpbrk found
+	 */
+	TEST_ASSERT_EQUAL_PTR(&loremIpsum[4], strpbrk(loremIpsum, "Ipsum"));
+
+	/* Like in the case before strpbrk found 'o' at second place in the array and his output point at second place in the array*/
+	TEST_ASSERT_EQUAL_PTR(&loremIpsum[1], strpbrk(loremIpsum, "Dolor"));
+
+	/* Simple usage of strpbrk with ascii set */
+	TEST_ASSERT_EQUAL_PTR(asciiStr, strpbrk(asciiStr, asciiStr));
+	TEST_ASSERT_EQUAL_PTR(&asciiStr[ASCII_LENGTH / 4], strpbrk(asciiStr, &asciiStr[ASCII_LENGTH / 4]));
+	TEST_ASSERT_EQUAL_PTR(&asciiStr[ASCII_LENGTH / 3], strpbrk(asciiStr, &asciiStr[ASCII_LENGTH / 3]));
+	TEST_ASSERT_EQUAL_PTR(&asciiStr[ASCII_LENGTH / 2], strpbrk(asciiStr, &asciiStr[ASCII_LENGTH / 2]));
+
+	free(asciiStr);
+}
+
+
+TEST(string_pbrk, empty_args)
+{
+	/* Different scenarios to acquire null pointer as output*/
+	TEST_ASSERT_NULL(strpbrk("", "abc"));
+	TEST_ASSERT_NULL(strpbrk("\0", "abc"));
+	TEST_ASSERT_NULL(strpbrk("\0", "\0"));
```

Comment:
Would be sufficient, if the first NUL term is detected, checking the next ones is not necessary.
```suggestion
	TEST_ASSERT_NULL(strpbrk("", ""));
```